src/base/conditions/BaseTextConditionRule.php

```diff
@@ -22,9 +22,9 @@ abstract class BaseTextConditionRule extends BaseConditionRule
     public string $operator = self::OPERATOR_EQ;
 
     /**
-     * @var string The input value.
+     * @var string|array The input value.
      */
-    public string $value = '';
+    public string|array $value = '';
```

Comment:
This feels like a breaking change. Think maybe Number rules should gain a `maxValue` property, which can be null when `between` isn’t used.